staging/src/k8s.io/apimachinery/pkg/labels/selector.go

```diff
@@ -132,6 +136,29 @@ type Requirement struct {
 //
 // The empty string is a valid value in the input values set.
 func NewRequirement(key string, op selection.Operator, vals []string) (*Requirement, error) {
+	return newRequirement(key, op, vals, validateLabelValue)
+}
+
+// NewFieldRequirement is the constructor for a Requirement of field.
+// If any of these rules is violated, an error is returned:
+// (1) The operator can only be In, NotIn, Equals, DoubleEquals, NotEquals, Exists, or DoesNotExist.
+// (2) If the operator is In or NotIn, the values set must be non-empty.
+// (3) If the operator is Equals, DoubleEquals, or NotEquals, the values set must contain one value.
+// (4) If the operator is Exists or DoesNotExist, the value set must be empty.
+// (6) The key is invalid due to its length, or sequence
+//     of characters. See validateLabelKey for more details.
+//
+// The empty string is a valid value in the input values set.
+func NewFieldRequirement(key string, op selection.Operator, vals []string) (*Requirement, error) {
```

Comment:
`pkg/fields/selector.go` contains field-selector related things... I didn't really expect this added to the labels package

Reply:
moved to `fields/selector.go`; but for now, we only support `equals` and `not equals`.